### DISCUSSION TITLE
Bugfixes for Queueables

### DISCUSSION THREAD

**Discussion by ghost** [dschibster/sfdx-batch-orchestrator#21] [Show and tell]
Hi dschibster,

I just fixed something and wanted you to know about it, maybe you want to find a fix for that issue yourself.

Failing queueables are "swallowed" for 2 reasons:

1. The exception will be caught by the finalizer execute in BatchJobBase and there it will be forgotten. In order to log the exception message, one option is this:

> public void execute(FinalizerContext ctx) {
>       if (ctx.getResult() == ParentJobResult.UNHANDLED_EXCEPTION) {
>           logInformation(ctx?.getAsyncApexJobId(), ctx.getException().getMessage());
>       }
>       finishBatch(ctx.getAsyncApexJobId());
>   }

  2. The logfiles attached to the "BatchJob" configuration will not pop up in the overview, since the Completion Date on failing Queueables will be null. One way to solve this is to change in BatchJobLogger:
  

> public static void updateParentQueueableLog(Id asyncJobId) {
>         AsyncApexJob job = [
>             SELECT
>                 Id,
>                 Status,
>                 NumberOfErrors,
>                 JobItemsProcessed,
>                 TotalJobItems,
>                 CompletedDate,
>                 CreatedBy.Email,
>                 ExtendedStatus
>             FROM AsyncApexJob
>             WHERE Id = :asyncJobId
>         ];
> 
>         Datetime completedDate = job.CompletedDate;
>         if (completedDate == null) {
>             completedDate = Datetime.now();
>         }
> 
>         BatchApexStatusEvent__e statusEvent = new BatchApexStatusEventBuilder()
>             .withContext('FINISH')
>             .withStatus(job.Status)
>             .withMessage(job.ExtendedStatus)
>             .withTimestamp(completedDate.getTime())
>             .withJobId(asyncJobId)
>             .getRecord();
> 
>         EventBus.publish(statusEvent);
>     }

    
    The outcome of this is: The batch jobs will always show up with a log and we get the full error texts.
    
    Sorry for not going the clean way of making a change request, I'm quite busy and didn't want this to be forgotten. Btw, any chance that this will be released as a managed package one day? We'd like to reduce our code load.
    
    BR,
    Christian

**Comment by dschibster** [comment]
Thank you Christian! I think I can add this to the existing code pretty quickly. I was about to ask for a Pull Request until I read to the end of the message. :)

Regarding Managed Packages: I may do that at some point in time, however I fear the security review. :)

**Reply by ghost** [reply]
damn you're fast

**Reply by dschibster** [reply]
Email Notifications are the shit. :D
